projects/packages/jetpack-mu-wpcom/src/features/calypso-locale-sync/calypso-locale-sync.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Sync Calypso locale when wp-admin locale is updated via /wp-admin/profile.php.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+if ( function_exists( 'wpcom_is_nav_redesign_enabled' ) && wpcom_is_nav_redesign_enabled() ) {
+	add_filter(
+		'insert_user_meta',
+		function ( $meta, $user, $update ) {
+			$locale     = $meta['locale'];
+			$old_locale = get_user_locale( $user );
+
+			if ( ! $update || $old_locale === $locale ) {
+				// Only for locale changes on an existing user
+				return $meta;
+			}
+
+			if ( ! $locale ) {
+				// No locale means the "Site Default" option which is the Site language (WPLANG).
+				$locale = get_option( 'WPLANG', '' );
+			}
+
+			Client::wpcom_json_api_request_as_user(
```

Comment:
Do we need to check whether the user is connected to a WP.com account before making this request?

Reply:
Good catch! I tested it. If we don't check for a wpcom connection, we make the API call and it responds with the WP_Error `missing_token`. The user wouldn't notice it, but we can prevent an unnecessary API call and its error response by checking if the user is connected to wpcom before making the call.